tests/models/test_hparams.py

```diff
@@ -642,3 +649,30 @@ def test_model_with_fsspec_as_parameter(tmpdir):
     )
     trainer.fit(model)
     trainer.test()
+
+
+@pytest.mark.skipif(not HYDRA_AVAILABLE, reason="Hydra is not available")
+def test_model_save_hyper_parameters_interpolation_with_hydra(tmpdir):
+
+    initialize(config_path="conf")
```

Comment:
initialize() is changing the global state.

Use this to avoid that.
```python
with initialize():
 ...
```

Please go over the [relevant page](https://hydra.cc/docs/experimental/compose_api#initialization-methods), there have been many changes in Hydra 1.0 you may not be aware of yet.

Reply:
Thanks Omry. I will adress those comments tomorrow 😁

Reply:
Hey Omry. How can I save / load a collection of normal dict, list and DictConfig, ListConfig ?

Reply:
I have tried to save without interpolation and reload with it, but I didn't manage to make it work ? Would you mind having a look ?

Reply:
For saving:
```
    if OMEGACONF_AVAILABLE:
        with fs.open(config_yaml, "w", encoding="utf-8") as fp:
            OmegaConf.save(hparams, fp)
```

```
args_1:
  cfg:
    log: ${training.log}
    training:
      log: Something
args_2:
- log: ${training.log}
  training:
    log: Something
kwarg_1:
  cfg:
  - log: ${training.log}
    training:
      log: Something
```

to reload (but it doesn't work as the root path depends on the structure saved.)
```
    if OMEGACONF_AVAILABLE:
        with fs.open(config_yaml, "r") as fp:
            hparams =  yaml.full_load(fp)
            for k, v in hparams.items():
                hparams[k] = OmegaConf.create(v)
            return hparams
```

Reply:
Current solution for saving

```
    if OMEGACONF_AVAILABLE:
        def resolve_dict_config(data):
            data = OmegaConf.to_container(data, resolve=True)
            return OmegaConf.create(data)

        hparams = apply_to_collection(hparams, DictConfig, resolve_dict_config)
        with fs.open(config_yaml, "w", encoding="utf-8") as fp:
            OmegaConf.save(hparams, fp)
        return
```

```
args_0:
  log: Something
  training:
    log: Something
args_1:
  cfg:
    log: Something
    training:
      log: Something
args_2:
- log: Something
  training:
    log: Something
kwarg_1:
  cfg:
  - log: Something
    training:
      log: Something
```